prime-router/src/test/kotlin/azure/ReportFunctionTests.kt

```diff
@@ -386,4 +437,49 @@ class ReportFunctionTests {
         verify(exactly = 1) { actionHistory.trackActionSenderInfo(any(), any()) }
         assert(resp.status.equals(HttpStatus.BAD_REQUEST))
     }
+
+    @Test
+    fun `test processRequest invalid Option`() {
+        // setup steps
+        val actionHistory = spyk(ActionHistory(TaskAction.receive))
+        var (reportFunc, req, sender) = setupForProcessRequestTests(actionHistory)
+        req.queryParameters["option"] = "INVALID OPTION"
+
+        // act on invalid options
+        var resp = reportFunc.processRequest(req, sender)
+
+        // No report is created and an error is returned to explain the issue
+        assert(resp.status.equals(HttpStatus.BAD_REQUEST))
+        verify(exactly = 1) { actionHistory.trackLogs(any<ActionLog>()) }
```

Comment:
I tried really hard to have a better validation that the error was generated here. I needed to make actionHistory a spy to mock out and prevent attempts to write to the db. Even if I didn't include line 149 `every { actionHistory.trackLogs(any<ActionLog>()) } returns Unit`, track logs was being mocked out and logs weren't being properly added even though when testing using postman I could clearly see the proper response. I'm very open to alternatives to this approach if there are ideas about a better way to test this.